### DISCUSSION TITLE
🎂 Shell commands turns one year! ❤

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#263] [Announcements]
# Happy birthday!

Today, on the 31th of August 2022, will be a year since the [_Shell commands_ plugin was accepted to Obsidian's community plugins list](https://github.com/obsidianmd/obsidian-releases/pull/433#event-5234028328).

The plugin started as a very small and simple project whose aim was to be able to execute system commands by pressing hotkeys in Obsidian. That has grown in the past year to include multiple ways to trigger execution of shell commands, ability to utilize outputs back in Obsidian, and many new variables for passing data from Obsidian.

## Prominent features implemented along the way
- Output channels for using text content received from shell commands in various ways, such as inserting into the current note, clipboard, or just displaying in the user interface.
- Special characters in `{{variable}}` values are safely escaped (except if shell is `CMD`).
- Autocompleting `{{variables}}` during typing.
- Shell commands can be executed via automatic events, context menus, and URI links.
- Prompts for asking values from user before executing a shell command.
- [See all improvements done up to `0.15.0`](https://github.com/Taitava/obsidian-shellcommands/blob/main/CHANGELOG.md#0150---2022-08-20).
- [See the most prominent future-coming features](https://publish.obsidian.md/shellcommands/Roadmap).

## Some stats until today
 - Commits (version `0.15.0`): 892
 - Stars: 106
 - Forks: 2
 - Watchers: 3
 - [Show and tell topics](https://github.com/Taitava/obsidian-shellcommands/discussions/categories/show-and-tell): 5

## Thank you!

I want to thank you @FelipeRearden for the huge amount of time and effort you have given for this project, by giving me feedback and ideas on how to improve this plugin! ❤️ I also want to thank everybody else who has contributed ideas and feedback, and @vrajur for developing the [{{caret_position}} variable](https://github.com/Taitava/obsidian-shellcommands/pull/118), and @dathanb for providing [an example on how to implement `stdin` support](https://github.com/Taitava/obsidian-shellcommands/pull/128) (which I still haven't implemented)! ❤️ All the input and contributions have made this plugin better than what I could have achieved alone! 🌻

**Comment by Taitava** [comment]
# 🥳  Two years of _Shell commands_ 🎉 

I forgot to write a two year celebration post on 2023-08-31, but I'll write some stats of today 2023-09-22.

## Prominent features along the way
- Support custom shells
- Realtime output handling
- Pass variables to stdin

## Some stats until today
- Commits (version [`0.20.0`](https://github.com/Taitava/obsidian-shellcommands/tree/0.20.0)): 1288
- Stars: 236 (⏫ Nice, doubled from last year ⏫ )
- Forks: 6
- Watchers: 6
- [Show and tell topics](https://github.com/Taitava/obsidian-shellcommands/discussions/categories/show-and-tell): 9
- Issues & discussions: 370

**Reply by dathanb** [reply]
Happy birthday!

**Comment by Taitava** [comment]
# 🎉 Three years now 🎈 

I feel this year went very fast and while I did do quite a big effort in developing the [debouncing system for event triggering](https://publish.obsidian.md/shellcommands/Events/Events+-+debouncing), I do feel that otherwise I haven't spent as much time developing this plugin further as I had done during the two previous years.

That said, it's still nice to see the plugin is actively used, there are new users coming and feedback and questions are posted. Thank you! ❤️  

## Prominent features along the way

- [Prompts: Support different field types](https://github.com/Taitava/obsidian-shellcommands/issues/365)
- [Event debouncing](https://github.com/Taitava/obsidian-shellcommands/issues/380)

## Some stats until today
 -   Commits (version [0.22.0](https://github.com/Taitava/obsidian-shellcommands/tree/0.22.0)): 1428
 -   Stars: 348 (⏫ Nice, +100 from last year ⏫ )
 -   Forks: 11
 -   Watchers: 6
 -   [Show and tell topics](https://github.com/Taitava/obsidian-shellcommands/discussions/categories/show-and-tell): 11
 -   Issues & discussions: 425
